Add content to working mode page (#9)
w3c/did-wg: WorkMode/guiding_principles.md

# DID Working Group Guiding Principles
{: .no_toc}

This document sets down the current guiding principles in use by the DID
Working Group to assist in making decisions about specification features.
These guidelines provide a framework in which to have productive
discussions that quickly come to consensus, rather than ending up blocked
in an unresolvable tug-of-war over opposing and equally valid points of
view. The principles are not meant to be interpreted as strict or objective
rules, but instead as informative of the overall direction of the group to
promote consistency and usability of the resulting specifications. 

This document is a *Living Document* and as such will change. Members of

> the group are encouraged to edit (e.g. to update, correct, etc.) the

Reviewer: ```suggestion
the group are encouraged to edit (e.g., to update, correct, etc.) the
```

## Stay on target!

We follow our overall mission to standardize
* the DID URI scheme, 
* the data model and syntax of DID Documents, 
* and the requirements for DID Method specifications.

**Notes**

* 

## Require real use cases, with actual instance data.

Features are supported by real world use cases, with actual data that can
be referenced. This helps reduce entirely theoretical features that might
be implemented in libraries, but never used in practice by producers or
consumers.

**Notes**

* Use cases should be supporting material which provides rationale for
    decisions, and a teaching tool for when features might be useful. They
    are not a design requirements gathering tool, and the group is not
    starting from an empty slate.
* These use cases are collected in a [Use Cases document](https://github.com/w3c/did-use-cases),
    which will be published as a Note.

## Require at least two supporters for each use case.

As the mission is interoperability, having at least two supporters for each
use case is important. If only one supporter can be found, then any
features that it would require are likely too specific for inclusion in the
proposed form.

**Notes**

> * Any independent organization, group or individual can be a supporter,

Reviewer: ```suggestion
* Any independent organization, group, or individual can be a supporter,
```

>     regardless of whether they are part of the Working Group, or members of

Reviewer: ```suggestion
    regardless of whether they are part of the Working Group, or a member of
```

    the W3C. Consortia or other standards organizations that represent many
    participants can thus fulfill this principle via their members.
* This is orthogonal to implementations of the specifications, which are
    required to exit Candidate Recommendation phase.

## As simple as possible, but no simpler.

> A simpler solution (to understand, implement and use) is better than a more

Reviewer: ```suggestion
A solution that is simpler (to understand, implement, and use) is better than a more
```

**Notes**

* Issues might arise where we need to prioritize between producers and
consumers of the DID. The general feeling is that, in that situation, we
should prioritize consumer happiness over producer happiness, i.e., the
usability of the data, rather than ease of creation.
  

## Provide on-ramps.

A solution that can be implemented in incremental stages is better than a
solution that is all or nothing, as not everyone needs every feature but
many people need various parts. Complex patterns should build upon simple
ones, sometimes making the patterns more complicated than if they were
built in isolation.

## Define success, not failure.

The specifications are defined in terms of what it means to be conformant,
and not patterns that are not conformant. The fewer constraints we require,
the easier to have non-breaking changes in the future and the easier it is
to have experimentation. In a similar way to the distinction between Open
World and Closed World, if something is not defined by the specification
then it is permissible (just not standardized) rather than being disallowed.

**Notes**

* It is hard to judge when it is appropriate to be prescriptive, and when

>     it is appropriate to be silent. Several issues have arisen on both

Author: ```suggestion
    it is appropriate to be silent.
```

>     sides -- sometimes 1.0 is too prescriptive, sometimes the definition is

Author: ```suggestion
```

>     not precise enough.

Reviewer 2: Not clear which 1.0 this is referencing.  Perhaps drop the final sentence?

>     not precise enough.

Author: ```suggestion
```
